src/m365/spo/commands/site/site-add.ts

```diff
@@ -526,152 +526,145 @@ class SpoSiteAddCommand extends SpoCommand {
-            }
-            else {
-              return Promise.resolve(false);
-            }
-          }
-        })
-        .then((): Promise<string> => {
-          if (this.verbose) {
-            logger.logToStderr(`Site doesn't exist. Checking if the site ${url} exists in the recycle bin...`);
-          }
+      const res1: any = await request.post(requestOptions);
+      const json: ClientSvcResponse = JSON.parse(res1);
+      const response: ClientSvcResponseContents = json[0];
 
-          const requestOptions: any = {
-            url: `${this.spoAdminUrl as string}/_vti_bin/client.svc/ProcessQuery`,
-            headers: {
-              'X-RequestDigest': (this.context as FormDigestInfo).FormDigestValue
-            },
-            data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="181" ObjectPathId="180" /><Query Id="182" ObjectPathId="180"><Query SelectAllProperties="true"><Properties /></Query></Query></Actions><ObjectPaths><Method Id="180" ParentId="175" Name="GetDeletedSitePropertiesByUrl"><Parameters><Parameter Type="String">${formatting.escapeXml(url)}</Parameter></Parameters></Method><Constructor Id="175" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
-          };
+      if (response.ErrorInfo) {
+        if (response.ErrorInfo.ErrorTypeName === 'Microsoft.Online.SharePoint.Common.SpoNoSiteException') {
+          const r = await this.siteExistsInTheRecycleBin(url, logger);
+          return r;
+        }
+        else {
+          throw response.ErrorInfo.ErrorMessage;
+        }
+      }
+      else {
+        const site: SiteProperties = json[json.length - 1];
 
-          return request.post(requestOptions);
-        })
-        .then((res: string): void => {
-          const json: ClientSvcResponse = JSON.parse(res);
-          const response: ClientSvcResponseContents = json[0];
-          if (response.ErrorInfo) {
-            if (response.ErrorInfo.ErrorTypeName === 'Microsoft.SharePoint.Client.UnknownError') {
-              resolve(false);
-            }
-            else {
-              reject(response.ErrorInfo.ErrorMessage);
-            }
-          }
-          else {
-            const site: DeletedSiteProperties = json[json.length - 1];
-            if (site.Status === 'Recycled') {
-              resolve(true);
-            }
-            else {
-              resolve(false);
-            }
-          }
-        }, (error: any): void => {
-          if (typeof error === 'boolean') {
-            resolve(error);
-          }
-          else {
-            reject(error);
-          }
-        });
-    });
+        if (site.Status === 'Recycled') {
+          throw true;
+        }
+        else {
+          return false;
+        }
+      }
+    }
+    catch (error) {
```

Comment:
Let's remove the entire try catch. It's not useful and unclear. We should not and need not throw booleans.